### ISSUE TITLE
Race condition between gitops and gitrepo controller

### ISSUE THREAD

**Issue by manno** [rancher/fleet#2435]
The [gitops gitjob controller](https://github.com/rancher/fleet/blob/main/internal/cmd/controller/gitops/reconciler/gitjob_controller.go#L174) needs RBAC from the [gitrepo controller](https://github.com/rancher/fleet/blob/main/internal/cmd/controller/reconciler/gitrepo_controller.go#L185-L187).

Move from gitrepo to gitops controller:
- [ ] `AuthorizeAndAssignDefaults`
- [ ] [rbac creation](https://github.com/rancher/fleet/blob/main/internal/cmd/controller/reconciler/gitrepo_controller.go#L167-L191) 
- [ ] helm values check
- [ ] `NewTargetsConfigMap`

Basically, only the display/summary status updates remain in the gitrepo controller. 
* SetStatusFromResourceKey
* UpdateDisplayState
* SetStatusFromBundles
* SetStatusFromBundleDeployments
* gitrepo.Status.Display.ReadyBundleDeployments = 

Can we improve the condition updates, so both controllers don't conflict on the same condition?

**Comment by manno** [comment]
Possible duplicate of https://github.com/rancher/fleet/issues/2343

**Comment by 0xavi0** [comment]
There is no longer race conditions between those 2 controllers, because https://github.com/rancher/fleet/pull/2537 merges the whole functionality into the `gitops` controller.
The `GitRepo` controller does not exist anymore.
